### PR TITLE
changefeed (ticdc): remove resolvedTs from etcd

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: ref #9058, ref #9235

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/owner/changefeed.go [hunk 1]
```diff
@@ -390,23 +391,23 @@ func (c *changefeed) tick(ctx cdcContext.Context, captures map[model.CaptureID]*
...
 			// We should keep the metrics updated even if the scheduler cannot
 			// advance the watermarks for now.
-			c.updateMetrics(currentTs, c.state.Status.CheckpointTs, c.state.Status.ResolvedTs)
+			c.updateMetrics(currentTs, c.state.Status.CheckpointTs, c.resolvedTs)
 		}
 		return nil
 	}
 
-	prevResolvedTs := c.state.Status.ResolvedTs
+	prevResolvedTs := c.resolvedTs
```
Comment (CharlesCheung96) [inline cdc/owner/changefeed.go:399]: Better to use `c.resolvedTs` directly.

File: cdc/model/changefeed.go [hunk 2]
```diff
@@ -584,3 +584,15 @@ func (t DownstreamType) String() string {
 	}
 	return "Unknown"
 }
+
+// ChangeFeedStatusForAPI uses to transfer the status of changefeed for API.
+type ChangeFeedStatusForAPI struct {
```
Comment (CharlesCheung96) [inline cdc/model/changefeed.go:589]: It is better to avoid creating duplicate entities here.

File: cdc/model/changefeed.go [hunk 3]
```diff
@@ -584,3 +584,15 @@ func (t DownstreamType) String() string {
...
+	// do not marshal ResolvedTs, so that will not be store in etcd
```
Comment (CharlesCheung96) [inline cdc/model/changefeed.go:590]:
```suggestion
```

File: cdc/model/changefeed.go (hunk 2 above)
Comment (asddongmen) [reply cdc/model/changefeed.go:589]: It is the best way I can find now...

File: cdc/model/owner.go [hunk 4]
```diff
@@ -269,8 +269,8 @@ func (p ProcessorsInfos) String() string {
 }
 
 // ChangeFeedStatus stores information about a ChangeFeed
+// It is stored in etcd.
 type ChangeFeedStatus struct {
```
Comment (CharlesCheung96) [inline cdc/model/owner.go:273]: Please add some comments to explain that `AdminJobType ` should be removed in future.

**Review by sdojjy — APPROVED** [review]

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**